### ISSUE TITLE
Wrong behaviors with Caching

### ISSUE THREAD

**Issue by twwhung** [williamw520/jsoda#3]
Hi! Currently I found there is a strange behavior with Caching .
Just take a look at below code.

```
     public MyModel{
    @Key
    public String id;

    public String attr1;
    public String attr2;

     }

    Jsoda jsoda = new Jsoda(credentials);
     // #1
     jsoda.query(MyModel.class).select("attr1").run();
     // #2
     MyModel myModel = jsoda.getDao(MyModel.class).get(id);
```

For #2, jsoda will get the item by caching. so attr2 will not be returned. 
I think in this situation jsoda should get the item from DB rather than from cache.

Thanks at advance!

**Comment by williamw520** [comment]
You are right.  The caching behavior is wrong.  Currently a partial entity object (fields via select) will be cached by its primary key, and a subsequent query of a full object with the same primary key will get back the partial object from cache.  This is a undesired behavior.

I'll put in a fix to partition the cache space by the primary key plus the hash signature of the selected fields.  That should cache the partial objects and the full objects separately and they won't interfere with each other.

Good catch.

**Comment by williamw520** [comment]
Fixed in version 0.13

**Comment by twwhung** [comment]
Hi!

Thanks for update. I will try the new one after weekend.
